cmd/kubeadm/app/preflight/checks.go

```diff
@@ -163,19 +164,49 @@ func (poc PortOpenCheck) Check() (warnings, errors []error) {
 	return nil, errors
 }
 
-// IsRootCheck verifies user is root
-type IsRootCheck struct{}
+// IsPrivilegedUserCheck verifies user is privileged (linux - root, windows - Administrator)
+type IsPrivilegedUserCheck struct{}
 
-// Check validates if an user has root privileges.
-func (irc IsRootCheck) Check() (warnings, errors []error) {
+// Check validates if an user has elevated (root or administrator) privileges.
+func (ipuc IsPrivilegedUserCheck) Check() (warnings, errors []error) {
 	errors = []error{}
-	if os.Getuid() != 0 {
-		errors = append(errors, fmt.Errorf("user is not running as root"))
+	var err error
+	if runtime.GOOS == "windows" {
+		err = ipuc.windowsPrivilegedCheck()
+	} else {
+		err = ipuc.linuxPrivilegedCheck()
+	}
+	if err != nil {
+		errors = append(errors, err)
 	}
 
 	return nil, errors
 }
 
+func (ipuc IsPrivilegedUserCheck) windowsPrivilegedCheck() error {
```

Comment:
What do you think about splitting this into more Go idiomatic way to `checks_windows.go` and have it build only in case of windows ?

Reply:
:+1: